### PR TITLE
Dual stack support

### LINKED ISSUE #175: network type parameter
### Describe the Feature

It is possible to pass `network_type` parameter - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster

image

### Expected Behavior

Support network_type parameter - `IPV4` or `DUAL`

### Use Case

Access clusters from ipv6-only networks

### Describe Ideal Solution

Support network_type parameter - `IPV4` or `DUAL`

### Alternatives Considered

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by ngoyal16**
## what

RDS cluster can be run in two network modes - IPV4 or DUAL.
Underlying module already supports this parameter

## why

It can be mandatory to enable it to be able to connect from ipv6 only runtimes

## references

Closes #175

### CODE REVIEW

**Comment by GabisCampana** [comment]
@Benbentwo

File: main.tf [hunk 1]
```diff
@@ -66,6 +77,17 @@ resource "aws_security_group_rule" "egress" {
...
 }
 
+resource "aws_security_group_rule" "egress_ipv6" {
+  count             = local.enabled && var.egress_enabled ? 1 : 0
+  description       = "Allow outbound ipv6 traffic"
+  type              = "egress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  cidr_blocks       = ["::/0"]
```
Comment (Benbentwo) [inline main.tf:87]: Tests for this seem to fail:
```
TestExamplesComplete 2024-09-06T18:00:04Z retry.go:99: Returning due to fatal error: FatalError{Underlying: error while running command: exit status 1; ╷
│ Error: "::/0" is not a valid IPv4 CIDR block
│ 
│   with module.rds_cluster.aws_security_group_rule.egress_ipv6,
│   on ../../main.tf line 87, in resource "aws_security_group_rule" "egress_ipv6":
│   87:   cidr_blocks       = ["::/0"]
│ 
╵}
```

Comment (ngoyal16) [reply main.tf]: @Benbentwo can you please rerun and check.

**Comment by ngoyal16** [comment]
@Benbentwo issue fixed. can you please try again

**Comment by Nuru** [comment]
/terratest

Previous tests failed do to infrastructure issues

**Review by Benbentwo — APPROVED** [review]

**Pull request merged**